tests-manager.js

```diff
@@ -0,0 +1,87 @@
+// ---------------------------------------------------------------------------
+// Usage: node tests-manager.js someNodeFile
+// ---------------------------------------------------------------------------
+
+"use strict";
+const fs = require('fs');
+const { spawn } = require('child_process');
+const util = require('node:util');
+const { exit } = require('process');
+const execFile = util.promisify(require('node:child_process').execFile);
+const log = require ('ololog').unlimited
+
+const TEST_TIMESTAMP = "./testTimestamp.json";
+
+const testFile = require(TEST_TIMESTAMP);
+const lastRun = testFile.lastRun;
+const sixHours = 60 * 60 * 6 * 1000;
+const currentTs = Date.now();
+
+if (process.argv.length < 3) {
+    log.red("Required node command not provided");
+    log.yellow("Example: node tests-manager.js run-tests-ws");
+    exit(1);
+}
+
+const COMMAND = process.argv[2];
+
+const argv = process.argv
+argv.splice(0,3);
+
+async function main() {
+
+    if (currentTs > sixHours + lastRun) {
+        // run all tests again
+        runCommand(argv)
+    } else {
+        const { stdout } = await execFile('git diff master', ['--name-only'], { shell:true });
+
+        if (stdout.length === 0) {
+            return;
+        }
+
+        const shouldRunEverything = stdout.indexOf ("Exchange") > -1 ||
```

Comment:
Just as a doubt, maybe there are other files beyond these predefined clauses, which might be missing? i.e. some changes  in root directory (say `keys.json` or any file in root directory), or something in dependencies or etc... In the past, I thought that more reliable way could be inverted logic -  "...if there is at least one file, that doesn't match derived exchange file path.."
so, iterate through changed files and find if there is even one occurrence where `filepath !== './js/${exchangeName}.js` then set flag to true. 
wdyt?

Reply:
thus, if there are only exchange-files present in pr, then test only those exchange-files, without needed to retest whole-things.

Reply:
ah, and missed, if there are `/examples/` occurrence, that can be also ignored as an exclusion.

Reply:
@ttodua the thing is, both whitelisting and blacklisting are tricky here, we don't want to run tests, for example if the docs or examples are updated, but we want if `run-tests` or `run-tests-ws` are, so I don't think there is a "one size fits all" rule here

Reply:
but why not blacklist instead of whitelist? i mean, it's easier to check  imo : `filepath !== './js/${exchangeName}.js && ! filepath.includes('/examples/) && ! filepath.includes('/docs/)`, because as I said, many things seem to be missing from the whitelisted (not only `run-tests` but other files that should cause full re-tests/re-build)